### PR TITLE
Block2db

### PR DESCRIPTION
**Pull request by adeelq**

### CODE REVIEW

File: Block/Recommended/Bestsellers.php [hunk 1]
```diff
@@ -7,81 +7,51 @@
 class Bestsellers extends \Magento\Catalog\Block\Product\AbstractProduct
```
Comment (cdiacon) [inline Block/Recommended/Bestsellers.php]: @adeelq bestsellers no products on the page

File: Model/ResourceModel/Catalog.php [hunk 2]
```diff
@@ -5,13 +5,250 @@
...
+
+    /**
      * Initialize resource.
      */
     public function _construct()
     {
         $this->_init('email_catalog', 'id');
     }
 
+    public function __construct(
```
Comment (cdiacon) [inline Model/ResourceModel/Catalog.php]: docblock missing

File: Model/ResourceModel/Wishlist.php [hunk 3]
```diff
@@ -52,4 +77,27 @@ public function resetWishlists($from = null, $to = null)
 
         return $num;
     }
+
+    /**
+     * @param $customerId
+     * @return bool
+     */
+    public function getWishlistFromCustomerId($customerId)
```
Comment (cdiacon) [inline Model/ResourceModel/Wishlist.php]: rename method to for ex. getWishlistsForCustomer

File: Block/Recommended/Bestsellers.php [hunk 4]
```diff
@@ -98,44 +68,10 @@ public function getLoadedProductCollection()
...
-        $productSkus = $reportProductCollection->getColumnValues('sku');
-
-        $productCollection = $this->productFactory->create()
-            ->getCollection()
-            ->addAttributeToSelect('*')
-            ->addFieldToFilter('sku', ['in' => $productSkus]);
-
-        return $productCollection;
+        return $this->catalogFactory->create()
+            ->getBestsellerCollection($from, $to, $limit, $storeId);
```
Comment (cdiacon) [inline Block/Recommended/Bestsellers.php]: @adeelq also good to move all the params : from, to, limit.. to model as it's used once ?!?

File: Block/Recommended/Bestsellers.php (hunk 1 above)
Comment (adeelq) [inline Block/Recommended/Bestsellers.php]: @cdiacon done

File: Block/Recommended/Bestsellers.php (hunk 4 above)
Comment (adeelq) [inline Block/Recommended/Bestsellers.php]: @cdiacon lets leave param checking in block.

File: Model/ResourceModel/Catalog.php [hunk 5]
```diff
@@ -167,7 +179,7 @@ public function getProductCollectionFromSku($productSkus, $limit = false)
     {
         $productCollection = [];
 
-        if (! empty($ids)) {
+        if (! empty($productSkus)) {
```
Comment (cdiacon) [inline Model/ResourceModel/Catalog.php]: rename $productSkus 
rename $getProductCollectionFromSku

**Pull request merged**